pipeline/load.py

```diff
@@ -31,42 +48,89 @@ def get_data_from_db(conn: connection, table: str, column = "*")-> pd.DataFrame:
 
-    return df
+    with conn.cursor() as cur:
+        cur.execute("""SELECT * FROM article where article_url = %s;""", [url])
+        duplicate = cur.fetchall()
+        if len(duplicate) > 0:
+            return None
+    return url
+
+
+def check_for_duplicate_authors(conn: connection, name: str) -> str:
+    """Checks for a duplicates in author table. Returns a dataframe without duplicates."""
+
+    with conn.cursor() as cur:
+        cur.execute("""SELECT * FROM author where author_name = %s;""", [name])
+        duplicate = cur.fetchall()
+        if len(duplicate) > 0:
+            return None
+    return name
 
 
 def add_to_article_table_from_pandas(conn: connection, df: pd.DataFrame) -> None:
     """Converts df into tuples, then adds to article table.
     NB: needs author column converted into foreign key reference"""
 
     with conn.cursor() as cur:
-        tuples = df.itertuples() # is this a list?
-        execute_values(cur, """INSERT INTO author (article_url, source, created_at,
-                       author) VALUES %s""", tuples)
-        cur.commit()
+        tuples = df.to_records(index=False)
+        execute_values(cur, """INSERT INTO article (article_url, source, created_at)
+                       VALUES %s;""", tuples)
+        conn.commit()
 
 
 def add_to_scraping_info_table_from_pandas(conn: connection, df: pd.DataFrame) -> None:
     """Converts df into tuples, then adds to scraping_info table.
     NB: needs article_id column converted into foreign key reference"""
 
     with conn.cursor() as cur:
-        tuples = df.itertuples() # is this a list?
+        tuples = df.to_records() # is this a list?
         execute_values(cur, """INSERT INTO scraping_info (scraped_at, title, body,
-                       article_id) VALUES %s""", tuples)
-        cur.commit()
+                       article_id) VALUES (%s)""", tuples)
+        conn.commit()
+
+# hardcode source
+
+def add_to_author_table(conn: connection, authors: pd.DataFrame) -> None:
+    """Converts df into tuples, then adds to authors table.
+    NB: needs article_id column converted into foreign key reference"""
+
+    with conn.cursor() as cur:
+        tuples = authors.itertuples()
+        execute_values(cur, """INSERT INTO author (author_name) VALUES %s;""", tuples)
+        conn.commit()
+
+
+def load_data():
+    """Complete data loading in one function. Used for main.py"""
+    db_conn = get_connected_locally()
+    print(db_conn)
+    df_transformed = pd.read_csv(TRANSFORMED_DATA)
+    print(df_transformed)
+
+    # removes duplicates
+    df_transformed["url"] = df_transformed["url"].apply\
+        (lambda x: check_for_duplicate_articles(db_conn, x))
+    df_transformed = df_transformed.dropna(subset=["url"])
+
+    # inserts articles
+    # df_for_article = df_transformed[["url", "author", "published"]].copy()
+    # df_for_article["author"] = "BBC"
+    # add_to_article_table_from_pandas(db_conn, df_for_article)
+
```

Comment:
Not sure commented out code pushed to main is good? Do we need this or can be be deleted??